regression/tests/helpers/api_clients.py

```diff
@@ -344,56 +345,56 @@ def get_url_from_email(self, matching_string):
         return get_target_url_from_text(matching_string, email_text)
 
 
-class EnrollmentApiClient(object):
+class EdxRestApiBaseClass(object):
     """
-    Enrollment API client
+    Base class for creating EdxRestAPiClient instance
     """
+    api_url_root = None
+    append_slash = True
 
-    def __init__(self, host=None, key=None):
-        self.host = host or ENROLLMENT_API_URL
-        self.key = key or ACCESS_TOKEN
+    def __init__(self):
+        assert self.api_url_root
+        access_token, __ = self.get_access_token()
+        self._client = EdxRestApiClient(
```

Comment:
`self._client` -> `self.client`

Reply:
hmm, I copied this code from ecommerce as is, I guess underscore is used to differentiate our custom client from any existing clients. Can you elaborate why we need to remove the underscore

Reply:
this method is not internal to this base class as in code below we are calling this so we should be using `client` IMHO

Reply:
I guess the underscore is used here because it is called only in derived classes and not available as public import, which is what the "weak internal use indicator" implies or I suppose it implies. Anyway will talk about it in person, would like to learn language's best practices from you as I am just a novice here

Reply:
The underscore was there because, in the context from which you copied, `_client` is a private variable.

Reply:
well that settles it, I removed the underscore. I may have misread the code but I thought the _client was being used in inherited classes as well just like I used it